eZ/Bundle/EzPublishCoreBundle/FieldType/RichText/Converter/Html5Input.php

```diff
@@ -22,4 +23,27 @@ public function __construct($stylesheet, ConfigResolverInterface $configResolver
         $customStylesheets = $customStylesheets ?: array();
         parent::__construct($stylesheet, $customStylesheets);
     }
+
+    /**
+     * Method checks if document contains any links with '[' or ']' sign an replaces it to its html entities version.
+     *
+     * @param DOMDocument $document
+     *
+     * @return DOMDocument
+     */
+    public function convert(DOMDocument $document)
+    {
+        // EZP-27562 - links containing [ or ] sign results in xml validation error
+        foreach ($document->getElementsByTagName('a') as $link) {
+            $href = $link->getAttribute('href');
+            if ($href !== '') {
+                $href = str_replace('[', '%5B', $href);
+                $href = str_replace(']', '%5D', $href);
```

Comment:
Sorry for the delay. I looked once again at this today.
> However php will understand both http://www.example.com?my_array[]=1&my_array[]=2 (unencoded) and http://www.example.com?my_array%5B%5D=1&my_array%5B%5D=2

> You need to cover query at least, fragment most probably and user / pass => this leads to piece of ugly code

This makes me think that we should just loop over `parse_url` result parts except for scheme, host and port. I'd assume that `/` in path is valid route separator and we could do such processing for every part (mostly to avoid some weird `if`s). This also requires our own custom function which will glue `$parts` back together as `http_build_url` from `pecl_http` does (I'm not comfortable adding an extra requirement on PECL extensions).

@tomaszmadeyski when you get back to this issue maybe try to go this direction and we'll see if this works for us (can't guarantee other reviewers will agree with this :-) )

// btw. I still have a weird feeling we're reinventing the wheel, but haven't found anything better to move this forward.